### PR TITLE
chore: migrate doc plugin to ts

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Migrates the documentation plugin to v5

### Why is it needed?

v5

### How to test it?

- Automated tests
- Testing the examples in https://docs.strapi.io/dev-docs/plugins/documentation

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/plugins/documentation/admin/src/components/SettingsForm.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    width: 1rem;
+    path {
+      fill: ${({ theme }) => theme.colors.neutral600};
+    }
+  }
+`;
+
+type SettingsFormProps = {
+  data?: DocumentInfos;
+  onSubmit: (body: unknown) => Promise<void>;
```
Comment (joshuaellis) [inline packages/plugins/documentation/admin/src/components/SettingsForm.tsx:47]: The body is the form values you pass the Formik.

File: packages/plugins/documentation/admin/src/pages/Settings.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { useIntl } from 'react-intl';
+
+import { SettingsForm } from '../components/SettingsForm';
+import { useGetInfosQuery, useUpdateSettingsMutation } from '../services/api';
+import { getTrad } from '../utils';
+
+const SettingsPage = () => {
+  const { toggleNotification } = useNotification();
+  const { formatMessage } = useIntl();
+  const { formatAPIError } = useAPIErrorHandler();
```
Comment (joshuaellis) [inline packages/plugins/documentation/admin/src/pages/Settings.tsx:14]: I think this should be using the _unstableFormatAPIError property, I don't think the errors are handled correctly if we use this one with Redux Toolkit's base query.

File: packages/plugins/documentation/admin/src/pages/Settings.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const { data, isError, isLoading } = useGetInfosQuery();
+  const [updateSettings] = useUpdateSettingsMutation();
+
+  const onUpdateSettings = async (body: any) => {
```
Comment (joshuaellis) [inline packages/plugins/documentation/admin/src/pages/Settings.tsx:18]: Should this handle server side validation errors or are we not that fussed for this?

**Review by joshuaellis — COMMENTED** [review]
Few comments, overall looks fine. I think there's some further refactoring to be done, I'm a bit hesitant though if we're handing this over to the community? Lmk, because I will write a ticket to do some nice clean ups if that's not the case 😄

**Comment by alexandrebodin** [comment]
> Few comments, overall looks fine. I think there's some further refactoring to be done, I'm a bit hesitant though if we're handing this over to the community? Lmk, because I will write a ticket to do some nice clean ups if that's not the case 😄

Considering the DX plans we will keep it for now, most likely deprecate in favor of the API playground so don't bother too much with this

File: packages/plugins/documentation/admin/src/pages/Settings.tsx (hunk 2 above)
Comment (alexandrebodin) [reply packages/plugins/documentation/admin/src/pages/Settings.tsx:14]: ho didn't know about that one ty

**Review by markkaylor — APPROVED** [review]
Looks good to me. I left a couple comments just to make sure the changes were intentional. I tested it locally.

File: packages/plugins/documentation/server/src/controllers/documentation.ts [hunk 4]
```diff
@@ -98,19 +115,21 @@ module.exports = {
...
 
     const { error } = ctx.query;
 
     try {
-      const layout = fs.readFileSync(path.join(__dirname, '..', 'public', 'login.html'));
-      const filledLayout = _.template(layout)({
-        actionUrl: `${strapi.config.server.url}${
-          strapi.config.get('plugin::documentation.x-strapi-config').path
-        }/login`,
+      // const layout = fs.readFileSync(path.join(__dirname, '..', 'public', 'login.html'));
```
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/plugins/documentation/server/src/controllers/documentation.ts:126]: Yes!

File: packages/plugins/documentation/server/src/controllers/documentation.ts [hunk 5]
```diff
@@ -55,10 +69,13 @@ module.exports = {
...
-        const layout = fs.readFileSync(
-          path.resolve(__dirname, '..', 'public', 'index.html'),
-          'utf8'
-        );
+
+        const layout = (await import('../public/index.html?raw')).default;
+        // fs.readFileSync(
+        //   path.resolve(__dirname, '..', 'public', 'index.html'),
+        //   'utf8'
+        // );
```
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/plugins/documentation/server/src/controllers/documentation.ts:77]: Yes!

File: packages/plugins/documentation/server/src/services/helpers/utils/clean-schema-attributes.ts [hunk 6]
```diff
@@ -159,88 +161,70 @@ const cleanSchemaAttributes = (
...
-              data: {
-                type: 'array',
-                items: componentSchemaRefName.length
-                  ? { $ref: `${componentSchemaRefName}ListResponseDataItemLocalized` }
-                  : {},
-              },
-            },
-          };
-          break;
-        }
```
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/plugins/documentation/server/src/services/helpers/utils/clean-schema-attributes.ts:212]: Yes we completely removed thos endpoints from v5 localization is managed directly via the base routes

**Pull request merged**